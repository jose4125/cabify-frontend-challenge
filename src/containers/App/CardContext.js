import React, { Component } from 'react';

export const CardContext = React.createContext();

class CardProvider extends Component {
  state = {
    fullName: '',
    jobDescription: '',
    prefix: '',
    phoneNumber: '',
    email: '',
    website: '',
    address: ''
  };

  render() {
    return (
      <CardContext.Provider
        value={{
          state: this.state,
          handleChange: (value, key) => this.setState({ [key]: value }),
          handlePrefix: prefix => {
            this.setState({ prefix });
          },
          handleReset: prefix => {
            this.setState({
              fullName: '',
              jobDescription: '',
              prefix: prefix,
              phoneNumber: '',
              email: '',
              website: '',
              address: ''
            });
          }
        }}
      >
        {this.props.children}
      </CardContext.Provider>
    );
  }
}

export default CardProvider;
