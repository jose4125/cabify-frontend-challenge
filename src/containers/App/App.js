import React from 'react';

import { CardContext } from './CardContext';
import Card from '../../components/Card';
import CardForm from '../CardForm';
import '../../styles/App.css';

const App = () => (
  <CardContext.Consumer>
    {context => (
      <div className="mainWrapper row">
        <Card info={context.state} />
        <CardForm context={context} />
      </div>
    )}
  </CardContext.Consumer>
);

export default App;
