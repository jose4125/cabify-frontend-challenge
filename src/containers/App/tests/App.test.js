import React from 'react';
import { shallow } from 'enzyme';

import { CardContext } from '../CardContext';
import App from '../App';
//import Card from '../../../components/Card';
//import CardForm from '../../CardForm';

describe('test', () => {
  const state = {
    fullName: '',
    jobDescription: '',
    prefix: '',
    phoneNumber: '',
    email: '',
    website: '',
    address: ''
  };

  const mockContext = {
    state: state
  };

  CardContext.Consumer = jest.fn(props => props.children({ ...mockContext }));

  const wrapper = shallow(<App />).dive();
  it('should render whit out crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a Card component', () => {
    expect(wrapper.find('Card').exists()).toBe(true);
  });

  it('should have a CardForm component', () => {
    expect(wrapper.find('CardForm').exists()).toBe(true);
  });
});
