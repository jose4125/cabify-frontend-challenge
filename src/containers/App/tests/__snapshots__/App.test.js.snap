// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should render whit out crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <mockConstructor>
    [Function]
  </mockConstructor>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          info={
            Object {
              "address": "",
              "email": "",
              "fullName": "",
              "jobDescription": "",
              "phoneNumber": "",
              "prefix": "",
              "website": "",
            }
          }
        />,
        <CardForm
          context={
            Object {
              "state": Object {
                "address": "",
                "email": "",
                "fullName": "",
                "jobDescription": "",
                "phoneNumber": "",
                "prefix": "",
                "website": "",
              },
            }
          }
        />,
      ],
      "className": "mainWrapper row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "info": Object {
            "address": "",
            "email": "",
            "fullName": "",
            "jobDescription": "",
            "phoneNumber": "",
            "prefix": "",
            "website": "",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "context": Object {
            "state": Object {
              "address": "",
              "email": "",
              "fullName": "",
              "jobDescription": "",
              "phoneNumber": "",
              "prefix": "",
              "website": "",
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            info={
              Object {
                "address": "",
                "email": "",
                "fullName": "",
                "jobDescription": "",
                "phoneNumber": "",
                "prefix": "",
                "website": "",
              }
            }
          />,
          <CardForm
            context={
              Object {
                "state": Object {
                  "address": "",
                  "email": "",
                  "fullName": "",
                  "jobDescription": "",
                  "phoneNumber": "",
                  "prefix": "",
                  "website": "",
                },
              }
            }
          />,
        ],
        "className": "mainWrapper row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "info": Object {
              "address": "",
              "email": "",
              "fullName": "",
              "jobDescription": "",
              "phoneNumber": "",
              "prefix": "",
              "website": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "context": Object {
              "state": Object {
                "address": "",
                "email": "",
                "fullName": "",
                "jobDescription": "",
                "phoneNumber": "",
                "prefix": "",
                "website": "",
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
  },
}
`;
